File: Java/src/main/java/com/gildedrose/ItemType.java
package com.gildedrose;

import java.util.Arrays;

public enum ItemType {
    AgedBrie("Aged Brie"),
    BackstagePass("Backstage passes to a TAFKAL80ETC concert"),
    Sulfuras("Sulfuras, Hand of Ragnaros"),
    Conjured("Conjured"),
    Normal("Normal");

    private final String name;

    ItemType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static ItemType fromName(String name) {
        if (name.startsWith("Conjured")) {
            return Conjured;
        }
        return Arrays.stream(ItemType.values())
            .filter(itemType -> itemType.getName().equals(name))
            .findFirst()
            .orElse(Normal);
    }
}

File: Kotlin/src/main/kotlin/com/platinumrose/ItemType.kt
package com.platinumrose

enum class ItemType(val value: String? = null) {
    AGED_BRIE("Aged Brie"),
    BACKSTAGE_PASSES("Backstage passes to a TAFKAL80ETC concert"),
    SULFURAS("Sulfuras, Hand of Ragnaros"),
    REGULAR;

    companion object {
        fun fromValue(value: String): com.platinumrose.ItemType {
            return entries.find { it.value == value } ?: com.platinumrose.ItemType.REGULAR
        }
    }
}